straxen/get_corrections.py

```diff
@@ -34,152 +35,76 @@ def correction_options_wrapped(run_id, conf, *arg):
-            f"Gain model {gain_model} resulted in a to_pe "
-            f"of length {len(to_pe)}, but n_pmts is {n_pmts}!")
-    return to_pe
+        # special case constant single value
+        elif 'constant' in model_conf:
+            if not isinstance(cte_value, (float, int)):
+                raise ValueError(f"User specify a model type {model_conf} "
+                                 "and should provide a number. Got: "
+                                 f"{type(cte_value)}")
+            correction = cte_value
+            return correction
 
+    elif isinstance(conf, tuple) and len(conf) == 3:
+        model_conf, global_version, is_nt = conf[:3]
+        cmt = straxen.CorrectionsManagementServices(is_nt=is_nt)
+        correction = cmt.get_corrections_config(run_id, conf[:2])
+        if correction.size == 0:
+            raise ValueError(f"Could not find a value for {model_conf} "
+                             f"please check it is implemented in CMT. ")
 
-FIXED_TO_PE = {
-    # just some dummy placeholder for nT gains
-    'gain_placeholder': np.repeat(0.0085, straxen.n_tpc_pmts),
-    # Gains which will preserve all areas in adc counts.
-    # Useful for debugging and tests.
-    'adc_tpc': np.ones(straxen.n_tpc_pmts),
-    'adc_mv': np.ones(straxen.n_mveto_pmts),
-    'adc_nv': np.ones(straxen.n_nveto_pmts)
-}
+        if 'to_pe_model' in model_conf:  # to_pe array
+            return correction
 
+        elif model_conf in corrections_w_file: # file's name (maps, NN, etc) 
+            correction = ' '.join(map(str, correction))
+            return correction
 
-@export
-@correction_options
-def get_correction_from_cmt(run_id, conf):
-    if isinstance(conf, str) and conf.startswith('https://raw'):
-        # Legacy support for pax files
-        return conf
-    if isinstance(conf, tuple) and len(conf) == 3:
-        is_nt = conf[-1]
-
-        model_type, global_version = conf[:2]
-        correction = global_version  # in case is a single value
-        if 'constant' in model_type:
-            if not isinstance(global_version, (float, int)):
-                raise ValueError(f"User specify a model type {model_type} "
-                                 "and should provide a number. Got: "
-                                 f"{type(global_version)}")
-        else:
-            cmt = straxen.CorrectionsManagementServices(is_nt=is_nt)
-            correction = cmt.get_corrections_config(run_id, conf[:2])
-            if correction.size == 0:
-                raise ValueError(f"Could not find a value for {model_type} "
-                                 "please check it is implemented in CMT. "
-                                 f"for nT = {is_nt}")
-
-        if 'samples' in model_type:
+        elif 'samples' in model_conf: # int baseline samples, etc
             return int(correction)
+
         else:
-            return float(correction)
+            return float(correction) # float elife, drift velocity, etc
```

Comment:
We could reduce the number of return statements here to a single one. This makes is slightly more readable. You could do something like 
```
if ....
    correction = ' '.join(map(str, correction))
elif ...... 
    correction = int(correction)
....
return correction 
```
In that case you can also remove if to_pe in ....

Reply:
Good suggestion, by the way we should be fine since we have all types: str, int, float and array